### ISSUE TITLE
Add BIP 158 (Neutrino) support

### ISSUE THREAD

**Issue by msgilligan** [bitcoinj/bitcoinj#1945]
Create an implementation and prepare a possible strategy to migrate from the default SPV client [BIP-0037](https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki) to [BIP-158](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki) (Neutrino)

**Comment by msgilligan** [comment]
.. and in my research I just found some comments by @schildbach here: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-July/017159.html

**Comment by java4tech** [comment]
I have looked at the link and noticed @schildbach mentioned 
> 2) It filters blocks only. It doesn't address unconfirmed transactions.

we can get unconfirmed transactions data from the mempool. As mentioned in the first point. Its even more resource incentive on mobile devices not to mention the network limitations around the world.

But for desktop its still a nice idea. If we decided to work on it then may be we should have it as another option along with BIP 37.
BIP 158 can be used in desktop and server environment. And BIP 37 can be used with light clients and also with trusted onion (for end-to end encryption) nodes.

**Comment by schildbach** [comment]
Yes, even a combination of BIP 158 (for blocks) and BIP 37 (for unconfirmed transactions) would be useful, even for mobile.

Hopefully BIP 158 filters will be committed to the blockchain not too far in the future.

**Comment by msgilligan** [comment]
I would like to start moving forward with adding this to **bitcoinj**.
